File: Examples/VisualBasic/SmartArts/AssistantNode.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Slides
Imports System

Namespace Aspose.Slides.Examples.VisualBasic.SmartArts
    Public Class AssistantNode
        Public Shared Sub Run()
            ' ExStart:AssistantNode
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_SmartArts()

            ' Creating a presentation instance
            Using pres As New Presentation(dataDir & "AssistantNode.pptx")
                'Traverse through every shape inside first slide
                For Each shape As IShape In pres.Slides(0).Shapes
                    ' Check if shape is of SmartArt type
                    If TypeOf shape Is Aspose.Slides.SmartArt.ISmartArt Then
                        'Typecast shape to SmartArtEx
                        Dim smart As Aspose.Slides.SmartArt.ISmartArt = CType(shape, Aspose.Slides.SmartArt.SmartArt)
                        'Traversing through all nodes of SmartArt shape

                        For Each node As Aspose.Slides.SmartArt.ISmartArtNode In smart.AllNodes
                            Dim tc As String = node.TextFrame.Text
                            ' Check if node is Assitant node
                            If node.IsAssistant Then
                                ' Setting Assitant node to false and making it normal node
                                node.IsAssistant = False
                            End If
                        Next node
                    End If
                Next shape
                ' Save Presentation
                pres.Save(dataDir & "ChangeAssitantNode_out.pptx", Aspose.Slides.Export.SaveFormat.Pptx)
            End Using
            ' ExEnd:AssistantNode
        End Sub
    End Class
End Namespace

File: Examples/CSharp/SmartArts/AssistantNode.cs
using System.IO;
using Aspose.Slides;
using System;

namespace Aspose.Slides.Examples.CSharp.SmartArts
{
    public class AssistantNode
    {
        public static void Run()
        {
            // ExStart:AssistantNode
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_SmartArts();

            // Creating a presentation instance
            using (Presentation pres = new Presentation(dataDir+ "AssistantNode.pptx"))
            {
                // Traverse through every shape inside first slide
                foreach (IShape shape in pres.Slides[0].Shapes)
                {
                    // Check if shape is of SmartArt type
                    if (shape is Aspose.Slides.SmartArt.ISmartArt)
                    {
                        // Typecast shape to SmartArtEx
                        Aspose.Slides.SmartArt.ISmartArt smart = (Aspose.Slides.SmartArt.SmartArt)shape;
                        // Traversing through all nodes of SmartArt shape

                        foreach (Aspose.Slides.SmartArt.ISmartArtNode node in smart.AllNodes)
                        {
                            String tc = node.TextFrame.Text;
                            // Check if node is Assitant node
                            if (node.IsAssistant)
                            {
                                // Setting Assitant node to false and making it normal node
                                node.IsAssistant = false;
                            }
                        }
                    }
                }
                // Save Presentation
                pres.Save(dataDir + "ChangeAssitantNode_out.pptx", Aspose.Slides.Export.SaveFormat.Pptx);
            }
            // ExEnd:AssistantNode
        }
    }
}